packages/demo/examples/TSDigisolPlatform/components/HierarchyComponent/index.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect } from "react";
+import { Text, View, TouchableOpacity, StyleSheet, Dimensions } from "react-native";
+
+import { SERVER_ENDPOINT } from "../../../../../../../../config/endpoint";
+
+export const Hierarchy = (props: {
+  appState: any;
+  label: any;
+  styles: any;
+  children: any;
+  setAppState: any;
+  layoutConfig: any;
+  setLayoutConfig: any;
+  getEvents: any;
+  events: any;
+}) => {
+  const {
+    appState,
+    label,
+    styles,
+    children,
+    setAppState,
+    layoutConfig,
+    setLayoutConfig,
+    getEvents,
+  } = props;
+
+  console.log(`label is ${label}`);
+  console.log(getEvents(`${label}-btn-one`, setLayoutConfig));
+  const [displayDataModal, setDisplayDataModal] = useState({"response":[{"catalogName":"loading","categoryName":"loading"}]});
+  const [key,setKey]=useState(0);
+
+  const fetchData = async () => {
+    const res = await fetch(
+      `${SERVER_ENDPOINT}v1/catalog/list`,
+      {
+        method: "POST",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          userId: "TsdAdmin",
+          roleKey: 1,
+          moduleName:"Catalog",
+          actionName:"Search",
+        }),
+      }
+    ).then((res) => {
+      return res.json()
+    })
+      .then((res) => {
+        console.log("required data in display ", res);
+        setDisplayDataModal(res)
+      });
+  };
+  const fetchCatData = async () => {
```

Comment:
hi @rajatkhade 

Move this code to helper component with some meaningfull function name

Reply:
This component is still under development